web/source/kmwbase.ts

```diff
@@ -28,6 +28,20 @@
    Copyright 2017-2019 SIL International
 ***/
 namespace com.keyman {
+
+  export interface OptionType {
+    root: string;
+    resources: string;
+    keyboards: string;
+    fonts: string;
+    attachType: undefined | 'auto' | 'manual';
+    ui: string;
+    setActiveOnRegister: string; // TODO: Convert to boolean
+
+    // Determines whether or not KeymanWeb should display its own alert messages
+    useAlerts: boolean;
```

Comment:
```suggestion
    root?: string;
    resources?: string;
    keyboards?: string;
    fonts?: string;
    attachType?: undefined | 'auto' | 'manual';
    ui?: string;
    setActiveOnRegister?: string; // TODO: Convert to boolean. Option loader needs to be able to receive this as a string or boolean

    // Determines whether or not KeymanWeb should display its own alert messages
    useAlerts?: boolean;
```

From https://help.keyman.com/developer/engine/web/15.0/reference/core/init#init_options
All these fields are optional, @jahorton ?